app/decorators/nav/link_builder.rb

```diff
@@ -0,0 +1,280 @@
+              permitted: h.policy(sample_period).index?,
+              icon: "folder-open"
+            }
+          ]
+        else
+          []
+        end
+      filter_and_set_active_nav_items(items, type: :sub, active: @context[:subsection])
+    end
+
+    def personal_nav_items
+      sample_export = Calendars::Exports::Export.new(user: h.current_user)
+      items =
+        [
+          {
+            name: :profile,
+            path: h.user_url(h.current_user),
+            permitted: h.policy(h.current_user).show?,
+            icon: "vcard"
+          }, {
+            name: :accounts,
+            path: h.yours_accounts_path,
+            permitted: h.policy(Billing::Account.new).yours?,
+            icon: "money",
+            i18n_key: multi_community? ? :accounts : :account
+          }, {
+            name: :calendars,
+            path: h.calendar_exports_path,
+            permitted: Calendars::ExportPolicy.new(h.current_user, sample_export).index?,
+            icon: "calendar"
+          }, {
+            name: :change_passwd,
+            path: h.people_password_change_path(h.current_user),
+            permitted: UserPolicy.new(h.current_user, h.current_user).update?,
+            icon: "asterisk"
+          }, {
+            name: :sign_out,
+            path: h.destroy_user_session_path,
+            permitted: true,
+            icon: "sign-out",
+            method: :delete
+          }
+        ]
+      filter_and_set_active_nav_items(items, type: :personal)
+    end
+
+    def footer_items
+      items =
+        case @context[:section]
+        when :wiki
+          [
+            {
+              name: :all,
+              parent: :wiki,
+              path: h.all_wiki_pages_path,
+              permitted: h.policy(Wiki::Page.new(community: h.current_community)).all?
+            }, {
+              name: :new,
+              parent: :wiki,
+              path: h.new_wiki_page_path,
+              permitted: h.policy(Wiki::Page.new(community: h.current_community)).create?
+            }
+          ]
+        else
+          []
+        end
+      filter_and_set_active_nav_items(items, type: :footer)
+    end
+
+    def filter_and_set_active_nav_items(items, type:, active: nil)
+      items.select! { |i| i[:permitted] }
+      items.each do |i|
+        i[:type] = type
+        i[:active] = true if active && i[:name] == active
+      end
+      items
+    end
+
+    def nav_link(item, tab: false)
+      params = {}
+      params[:method] = item[:method]
+      params[:role] = "tab" if tab
+      params[:"aria-controls"] = item[:name] if tab
+
+      i18n_sub_key = item[:parent] ? "#{item[:parent]}." : ""
+      name = t("nav_links.#{item[:type]}.#{i18n_sub_key}#{item[:i18n_key] || item[:name]}")
+
+      icon = item[:icon] ? h.icon_tag(item[:icon]).dup << " " : ""
+
+      h.link_to(icon.dup << " #{name}", item[:path], params)
+    end
+
+    def lens_path_if_present(controller, index_path: nil)
```

Comment:
I believe this method and `filter_and_set_active_nav_items` can be private. If a method can be private or protected (i.e. it's not needed outside the class or outside subclasses) it should be.

Reply:
Didn't work.  lens_path_if_present is called from _footer.erb.  Not sure where filter_and_set_active_nav_items is called, but rspec gives an error when I protect the method.

Reply:
Right you are on `lens_path_if_present`. `filter_and_set_active_nav_items` should be fine (you can do a global code search to confirm), but it's not a big deal.

Reply:
A search on filter_and_set_active_nav_items turned up nothing.  I protected it and don't see any errors.